ml-algorithms/src/main/java/org/opensearch/ml/engine/MultiTenantPredictable.java

```diff
@@ -0,0 +1,18 @@
+package org.opensearch.ml.engine;
+
+import java.util.Map;
+
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.engine.encryptor.Encryptor;
+
+public interface MultiTenantPredictable extends Predictable {
+
+    /**
+     * Init model (load model into memory) with ML model content and params.
+     * @param model ML model
+     * @param params other parameters
+     * @param encryptor encryptor
+     * @param tenantId tenantId
+     */
+    void initModel(MLModel model, Map<String, Object> params, Encryptor encryptor, String tenantId);
```

Comment:
Not sure why we need this new dedicated interface only for multi-tenant. Could we just include the tenantId as a parameter in the existing params Map?

Reply:
Good call out. Yeah Initially I wanted to have a separate interface for multi-tenancy but then I ended up having tenantId in the Encryptor class. I will remove this interface class in the next follow up PR if that's ok.

Reply:
Sure, I am OK.